### PR TITLE
Charts: Add support for legends and grids

### PR DESCRIPTION
**Pull request by dlabrecq**
Similar to the new donut utilization charts, this adds legend support for charts like area, bar, donut, line, pie, & stack. This helps users by automatically centering the legend within the same SVG, both vertically and horizontally.

In addition, this adds support to show grid and ticks along one axis, per the PF4 design below.
https://drive.google.com/file/d/1jCxBH0niB78V0PJvzHDBr8Hm5wKmDlgH/view

Fixes:
https://github.com/patternfly/patternfly-react/issues/2075
https://github.com/patternfly/patternfly-react/issues/2323
https://github.com/patternfly/patternfly-react/issues/1692

Screen Shot 2019-06-21 at 12 41 11 PM
Screen Shot 2019-06-21 at 12 41 24 PM
Screen Shot 2019-06-21 at 12 41 34 PM
Screen Shot 2019-06-21 at 12 41 52 PM
Screen Shot 2019-06-21 at 12 42 12 PM

### CODE REVIEW

**Comment by gdoyle1** [comment]
@dlabrecq Those gridlines should be behind the chart fills if that's possible!

**Comment by dlabrecq** [comment]
@gdoyle1 Moved the grid lines, in the examples, so they are rendered behind the charts

File: packages/patternfly-4/react-charts/src/components/ChartVoronoiContainer/ChartVoronoiContainer.tsx [hunk 1]
```diff
@@ -122,7 +124,13 @@ export const ChartVoronoiContainer: React.FunctionComponent<ChartVoronoiContaine
   theme = getTheme(themeColor, themeVariant),
   labelComponent = <ChartTooltip theme={theme} />,
   ...rest
-}: ChartVoronoiContainerProps) => <VictoryVoronoiContainer labelComponent={labelComponent} {...rest} />;
+}: ChartVoronoiContainerProps) => {
+  // Note: theme is required by voronoiContainerMixin, but VictoryVoronoiContainer is missing a prop type
+
+  // @ts-ignore
```
Comment (redallen) [inline packages/patternfly-4/react-charts/src/components/ChartVoronoiContainer/ChartVoronoiContainer.tsx]: That's neat.

File: packages/patternfly-4/react-charts/src/components/ChartVoronoiContainer/ChartVoronoiContainer.tsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
 import * as React from 'react';
 import hoistNonReactStatics from 'hoist-non-react-statics';
 import {
+  createContainer,
+  VictoryContainer,
```
Comment (TheRealJon) [inline packages/patternfly-4/react-charts/src/components/ChartVoronoiContainer/ChartVoronoiContainer.tsx]: Looks like these are imported but never used?

Comment (dlabrecq) [reply packages/patternfly-4/react-charts/src/components/ChartVoronoiContainer/ChartVoronoiContainer.tsx]: Thanks, removed the unused imports.

**Comment by redallen** [comment]
Conflicts from merging #2341. If those are resolved and @TheRealJon approves, we can get this in today.

**Comment by dlabrecq** [comment]
Fixed the merge conflict.

**Comment by TheRealJon** [comment]
@dlabrecq Looks like you missed a conflict marker at `ChartDonutUtilization.tsx:L15`

**Comment by dlabrecq** [comment]
Fixed

**Comment by TheRealJon** [comment]
@dlabrecq Can you squash before we merge this?

**Pull request merged**